tests/tests/test_management_config.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2021 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import uuid
+import pytest
+
+from common import management_api_with_params
+from internal_api import InternalAPIClient
+
+
+@pytest.fixture
+def device_id():
+    client = InternalAPIClient()
+    device_id = str(uuid.uuid4())
+    new_device = {"device_id": device_id}
+    r = client.provision_device_with_http_info(
+        tenant_id="", new_device=new_device, _preload_content=False
+    )
+    assert r.status == 201
+    yield device_id
+    r = client.decommission_device_with_http_info(
+        tenant_id="", device_id=device_id, _preload_content=False
+    )
+    assert r.status == 204
+
+
+class TestManagementConfig:
+    def test_config_device(self, device_id):
+        user_id = str(uuid.uuid4())
+        client = management_api_with_params(user_id=user_id)
+        #
+        # get the configuration (empty)
+        r = client.get_key_value_pair_store(device_id)
+        data = r.to_dict()
+        assert {"id": device_id, "reported": {}, "configured": {},} == {
+            k: data[k] for k in ("id", "reported", "configured")
+        }
+        assert "updated_ts" in data.keys()
+        #
+        # set the initial configuration
+        configuration = {
+            "key": "value",
+            "another-key": "another-value",
+            "dollar-key": "$",
+        }
+        r = client.store_a_set_of_key_value_pairs_with_http_info(
+            device_id, request_body=configuration, _preload_content=False
+        )
+        assert r.status == 204
+        #
+        # get the configuration
+        r = client.get_key_value_pair_store(device_id)
+        data = r.to_dict()
+        assert {
+            "id": device_id,
+            "reported": {},
+            "configured": {
+                "key": "value",
+                "another-key": "another-value",
+                "dollar-key": "$",
+            },
+        } == {k: data[k] for k in ("id", "reported", "configured")}
+        assert "updated_ts" in data.keys()
+        #
+        # replace the configuration
```

Comment:
I would love to see two more cases:
1. from the existing configuration set on key to empty value, so:
* first we PUT:
```json
configuration = {
            "key": "value",
            "another-key": "another-value",
        }
```
* we check that it is there with GET
* we put empty value:
```json
configuration = {
            "key": "value",
            "another-key": "",
        }
```
* we verify with GET that it is returned as empty.

2. what happens when you PUT empty config:
```json
configuration = {
        }
```

Reply:
@merlin-northern I implemented (1), (4) and (5); (3) is already in place, we always check if reported is empty, in all the tests. (2) is missing from your list :)